### PR TITLE
Switch to using LibCURL_jll / MozillaCACerts_jll

### PR DESCRIPTION
**Pull request by omus**
I used FTPClient.jl with these changes and the tests there did pass

### CODE REVIEW

**Comment by omus** [comment]
CI failure on Linux is returning `CURLE_SSL_CACERT`. This failure was [previously expected](https://github.com/JuliaWeb/LibCURL.jl/blob/2fc4266c8092bb712b6bd0ab5f2f1c9290294870/test/ssl.jl#L33) on FreeBSD, macOS, and Windows but it is now also happening on Linux.

cc: @staticfloat

**Comment by omus** [comment]
Related to: https://github.com/JuliaWeb/LibCURL.jl/pull/57

**Comment by staticfloat** [comment]
Can you re-run CI and see if it fails with the latest LibCURL jll?

**Comment by ViralBShah** [comment]
Does the Julia build pull libcurl from this jll? Just like `openblas_jll`, can we have packages that need `libcurl_jll` to simply use the one that ships with Julia itself?

**Comment by staticfloat** [comment]
That's part of what https://github.com/JuliaLang/julia/pull/35193 will provide

**Comment by aviks** [comment]
I submitted a change to LibCURL_jll to store the mozilla ca bundle. It'll require some change here which I'll do once the jll update is pushed and registered, but hopefully that'll solve the cert issues here (and at many other places)

**Comment by omus** [comment]
I've pulled in the commits from #78 to get this PR to pass (I don't have access to update the branch for #78)

**Comment by omus** [comment]
Everything looks good now. I want to validate these changes with FTPClient.jl but I'll need to do that tomorrow.

File: Project.toml [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-BinaryProvider = "b99e7846-7c00-51b0-8f62-c81ae34c0232"
-Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
+LibCURL_jll = "deac9b47-8bc7-5906-a0fe-35ac56dc84c0"
+MozillaCACerts_jll = "14a3606d-f60d-562e-9121-12d972cd8159"
 
 [compat]
-BinaryProvider = "0.5"
-julia = "0.7, 1.0"
+LibCURL_jll = "7.70"
+MozillaCACerts_jll = "2020"
```
Comment (giordano) [inline Project.toml:11]: If you want to always allow the latest certificate, I _think_ automerge is happy if you don't specify a compat bound for a jll package, but I need to double check this claim

**Comment by aviks** [comment]
> I want to validate these changes with FTPClient.jl

Yeah, it's worth checking that the jll binary does ftp correctly.  (BTW, it should also do http2 now)

I tested these changes with SMTPClient.jl (which also had numerous reports of certificate problems), and they are all fixed with this change, waiting for a tag here.

File: Project.toml [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 [deps]
-BinaryProvider = "b99e7846-7c00-51b0-8f62-c81ae34c0232"
-Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
+LibCURL_jll = "deac9b47-8bc7-5906-a0fe-35ac56dc84c0"
+MozillaCACerts_jll = "14a3606d-f60d-562e-9121-12d972cd8159"
 
 [compat]
-BinaryProvider = "0.5"
-julia = "0.7, 1.0"
+LibCURL_jll = "7.70"
```
Comment (omus) [inline Project.toml:10]: @aviks is the minimum bump required? I believe we're safe to leave the minimum as 7.68. Reference: https://github.com/JuliaWeb/LibCURL.jl/pull/78#discussion_r419125119

File: Project.toml (hunk 1 above)
Comment (omus) [reply Project.toml:11]: I just added this as it seemed best to have a minimum defined. I will admit this minimum is rather arbitrary

Comment (giordano) [reply Project.toml:11]: The fact is that this isn't just a minimum but also a maximum: only this year certificates will work with this release

File: Project.toml (hunk 2 above)
Comment (geokollias) [reply Project.toml:10]: @omus I think it's required to pick up HTTP/2 support (`nghttp2`)

File: Project.toml (hunk 1 above)
Comment (omus) [reply Project.toml:11]: Ah yes, you're right as this is working as a major version number. I'll update to: `>=2020`

File: Project.toml (hunk 2 above)
Comment (giordano) [reply Project.toml:10]: Having 7.68 here allows installing 7.70 as well

Comment (omus) [reply Project.toml:10]: Ok, I'll change that back as there's nothing specifically in LibCURL.jl that requires it.

**Comment by omus** [comment]
> I want to validate these changes with FTPClient.jl

Changes work fine with FTPClient.jl. I did notice [a failure](https://github.com/invenia/FTPClient.jl/issues/91) but I can reproduce the failure without this PR.

File: Project.toml (hunk 2 above)
Comment (geokollias) [reply Project.toml]: Good to know. So if a package depends on LibCURL.jl and wants HTTP/2 support as well, the Package.toml should look like:

```
[deps]
LibCURL = "..."
LibCURL_jll = "..."

[compat]
LibCURL_jll = "7.70"
```
?

Comment (giordano) [reply Project.toml]: @geokollias If you want to require at least version 7.70 of `LibCURL_jll` for your package, yes.  Remember that for automerge in General you need upper bounds in compat for all (non-JLL) packages :wink:

**Pull request merged**